ci/Jenkinsfiles/build.groovy

```diff
@@ -295,7 +295,9 @@ def buildUnitTestStage(env) {
             }
           } catch(err) {
             // TODO NXP-29512: workaround to know which env is failing later on
-            throw new Exception("${env}: ${err.message}", err)
+            def errorMessage = "${env}: ${err.message}"
+            echo "Throwing error with message: ${errorMessage}"
+            error "${errorMessage}"
```

Comment:
https://www.jenkins.io/doc/pipeline/steps/workflow-basic-steps/#error-error-signal
sais error aborts the build and avoids printing the stack trace. But since you are printing the error message anyway, why not sticking with the one-liner throw?

Reply:
The reason of the change is all explained in the [commit message](https://github.com/nuxeo/nuxeo/pull/4280/commits/63db157b19a4f050fcbab3ccee62d6848e7eeaf5).

The `echo` here is for debugging purpose, to output the custom message that we're building.

The error will then be thrown and catched by the surrounding `try/catch` block: https://github.com/nuxeo/nuxeo/blob/fix-NXP-29596-fail-PR-build-on-dev-unit-tests-failure/ci/Jenkinsfiles/build.groovy#L540, that might decide to continue or to rethrow, in which case the error message will be finally printed at the end of the build logs.

Reply:
Ok I get it. So this echo will be done as long as we catch and if in dev environment, the message will be printed at the end of the logs to facilitate seeing  it

Reply:
If not on a PR and in dev yes.
The error is then simply propagated as for any kind of failure, so the pipeline will naturally make the build fail and display the error message at the end of the logs.